website/project/model.py

```diff
@@ -75,16 +74,9 @@ def has_anonymous_link(node, auth):
     :param str link: any view-only link in the current url
     :return bool anonymous: Whether the node is anonymous to the user or not
     """
-    view_only_link = auth.private_key or request.args.get('view_only', '').strip('/')
-    if not view_only_link:
-        return False
-    if node.is_public:
-        return False
-    return any(
-        link.anonymous
-        for link in node.private_links_active
-        if link.key == view_only_link
-    )
```

Comment:
`private_links_active` is [a computed attribute](https://github.com/alexschiller/osf.io/blob/f61081441bca96f4f1d83a1e88ffa4d7d5184caa/website/project/model.py#L905-L907) that queries the `privatelink` collection for documents that are not `is_deleted`. The effect of the `return any`, then, is to search for the requested `privatelink.key` (that's what `view_only_link` is) amongst the non-deleted `privatelink`s. `private_links_active` dereferences another computed attribute, `private_links`, which in turn accesses `privatelink__shared`. I don't find another instance of `privatelink__shared` in the codebase; my hunch is that the `__shared` suffix is meaningful to our home-grown ODM.